toolchain/lexer/token_kind.h

```diff
@@ -6,38 +6,22 @@
 #define CARBON_TOOLCHAIN_LEXER_TOKEN_KIND_H_
 
 #include <cstdint>
-#include <initializer_list>
-#include <iterator>
 
 #include "common/ostream.h"
-#include "llvm/ADT/StringRef.h"
+#include "toolchain/common/enum_base.h"
 
 namespace Carbon {
 
-class TokenKind {
-  // Note that this must be declared earlier in the class so that its type can
-  // be used, for example in the conversion operator.
-  enum class KindEnum : int8_t {
-#define CARBON_TOKEN(TokenName) TokenName,
+CARBON_DEFINE_RAW_ENUM_CLASS(TokenKind, uint8_t){
+#define CARBON_TOKEN(Name) CARBON_RAW_ENUM_ENUMERATOR(Name)
```

Comment:
FWIW, in my version of this I renamed all of the `TokenName` usages elsewhere to be `Name` for consistency.

I'm happy to land this as-is and do a follow-up to rename the rest though.

Reply:
Switched for consistency.